plugin/evm/mempool.go

```diff
@@ -266,9 +280,37 @@ func (m *Mempool) addTx(tx *Tx, force bool) error {
 	// and CancelCurrentTx.
 	m.newTxs = append(m.newTxs, tx)
 	m.addPending()
+
+	m.bloom.Add(&GossipAtomicTx{Tx: tx})
+	_ = gossip.ResetBloomFilterIfNeeded(m.bloom, gossip.DefaultBloomMaxFilledRatio)
```

Comment:
This is done outside of `Add` so that the caller can choose how to refresh this. I think for peer lists (which are not as dynamic/changing constantly), we will refresh these much less frequently/maybe introduce a time heuristic.

Reply:
Sorry if I was unclear, I meant adding the bloom filter logic (add and reset) to `mempool.Add` (which is a new function) vs `mempool.addTx` (which is an existing one)

Reply:
Ah I see, sorry. Yeah I added this to `addTx` because we don't necessarily always add the tx to the mempool in some cases. There are some cases where we just drop the transaction in `addTx`.